core/field_textarea.js

```diff
@@ -0,0 +1,217 @@
+ * dereferencing any string table references.
+ * @param {!Object} options A JSON object with options (text, class, and
+ *                          spellcheck).
+ * @return {!Blockly.FieldTextArea} The new field instance.
+ * @package
+ * @nocollapse
+ */
+Blockly.FieldTextArea.fromJson = function(options) {
+  var text = Blockly.utils.replaceMessageReferences(options['text']);
+  var field = new Blockly.FieldTextArea(text, options['class']);
+  if (typeof options['spellcheck'] === 'boolean') {
+    field.setSpellcheck(options['spellcheck']);
+  }
+  return field;
+};
+
+/**
+ * Get the text from this field as displayed on screen.  May differ from getText
+ * due to ellipsis, and other formatting.
+ * @return {string} Currently displayed text.
+ * @private
+ */
+Blockly.FieldTextArea.prototype.getDisplayText_ = function() {
+  var value = this.value_;
+  if (!value) {
+    // Prevent the field from disappearing if empty.
+    return Blockly.Field.NBSP;
+  }
+  value = value.split("\n").map(function(value) {
+    if (value.length > this.maxDisplayLength) {
+      // Truncate displayed string and add an ellipsis ('...').
+      value = value.substring(0, this.maxDisplayLength - 4) + '...';
+    }
+    // Replace whitespace with non-breaking spaces so the value doesn't collapse - keep newline.
+    value = value.replace(/\s/g, Blockly.Field.NBSP);
+    return value;
+  }.bind(this)).join("\n");
+  if (this.sourceBlock_.RTL) {
+    // The SVG is LTR, force value to be RTL.
+    value += '\u200F';
+  }
+  return value;
+};
+
+/**
+ * Updates the colour of the htmlInput given the current validity of the
+ * field's value.
+ * @protected
+ */
+Blockly.FieldTextArea.prototype.render_ = function() {
+  if (!this.visible_) {
+    this.size_.width = 0;
+    return;
+  }
+  // Replace the text.
+  var textElement = this.textElement_;
+  textElement.setAttribute('class', 'blocklyText blocklyTextCode');
+  goog.dom.removeChildren(/** @type {!Element} */ (textElement));
+  var txt = this.getDisplayText_();
+  var y = 0;
+  var yoffset = 14; // 12.5 is hard-coded in Blockly.Field
+  var txtLines = txt.split("\n");
+  txtLines.forEach(function(t) {
+    Blockly.utils.dom.createSvgElement('tspan', {x:0,y:y + yoffset}, textElement)
+        .appendChild(document.createTextNode(t));
+    y += 20;
+  });
+  if (txtLines.length == 0) {
+    y += 20;
+  }
+
+  // set up widths
+  this.size_.width = this.textElement_.getBBox().width + 5;
+  this.size_.height = y + (Blockly.BlockSvg.SEP_SPACE_Y + 5) ;
+
+  if (this.borderRect_) {
+    this.borderRect_.setAttribute('width',
+        this.size_.width + Blockly.BlockSvg.SEP_SPACE_X);
+    this.borderRect_.setAttribute('height',
+        this.size_.height - (Blockly.BlockSvg.SEP_SPACE_Y + 5));
+  }
+};
+
+/**
+ * Create the text input editor widget.
+ * @return {!HTMLInputElement} The newly created text input editor.
+ * @private
+ */
+Blockly.FieldTextArea.prototype.widgetCreate_ = function() {
```

Comment:
I don't think this will work, unfortunately. FieldTextArea creates a `textarea` element, while FieldTextInput creates an `input` element. I could refactor widgetCreate_ to create its element based on a field, but that feels outside my scope.